docs/docs/aztec/concepts/accounts/keys.md

```diff
@@ -3,157 +3,128 @@ title: Keys
+All key pairs are derived using elliptic curve public-key cryptography on the [Grumpkin curve](https://github.com/AztecProtocol/aztec-connect/blob/9374aae687ec5ea01adeb651e7b9ab0d69a1b33b/markdown/specs/aztec-connect/src/primitives.md). Where the secret key is represented as a scalar and the public key is represented as an elliptic curve point multiplied by that scalar.
 
-#### Using shared state
+The address private key is an exception and derived in a way described above in the section “Address keys”.
 
-A compromise between the two solutions above is to use shared state. This would not generate additional nullifiers and commitments for each transaction while allowing the user to rotate their key. However, this causes every transaction to now have a time-to-live determined by the frequency of the mutable shared state, as well as imposing restrictions on how fast keys can be rotated due to minimum delays.
+### The special case of escrow contracts
 
-#### Reusing some of the in-protocol keys
+Typically, for account contracts the public keys will be non-zero and for non-account contracts zero.
 
-It is possible to use some of the key pairs defined in protocol (e.g. incoming viewing keys) as the signing key.
-Since this key is part of the address preimage (more on this in the privacy master key section), it can be validated against the account contract address rather than having to store it.
-However, this approach is not recommended since it reduces the security of the user's account.
+An exception (a non-account contract which would have some of the keys non-zero) is an escrow contract. Escrow contract is a type of contract which on its own is an "owner" of a note meaning that it has a` Npk_m` registered and the notes contain this `Npk_m`.
 
-#### Using a separate keystore
+Participants in this escrow contract would then somehow get a hold of the escrow's `nsk_m` and nullify the notes based on the logic of the escrow. An example of an escrow contract is a betting contract. In this scenario, both parties involved in the bet would be aware of the escrow's `nsk_m`. The escrow would then release the reward only to the party that provides a "proof of winning".
 
-Since there are no restrictions on the actions that an account contract may execute for authenticating a transaction (as long as these are all private function executions), the signing public keys can be stored in a [separate keystore contract](https://vitalik.ca/general/2023/06/09/three_transitions.html) that is checked on every call. This will incur in a higher proving time for each transaction, but has no additional cost in terms of fees, and allows for easier key management in a centralized contract.
+### App siloed keys
 
-### Complete address
+All keys on Aztec (except for the signing keys) are app-siloed meaning they are scoped to the contract that requests them. This means that the keys used for the same user in two different application contracts will be different.
```

Comment:
Address keys are also not app-siloed, right? So its just incoming view keys and nullifier keys?